pkg/platform/kube/resourcescaler/resourcescaler.go

```diff
@@ -395,3 +412,42 @@ func (n *NuclioResourceScaler) verifyReadiness(ctx context.Context, function *nu
 	}
 	return nil
 }
+
+// ResolveTargetsFromIngressCallback is scalertype.ResolveTargetsFromIngressCallback callback that extracts
+// Nuclio function target names from an Ingress, supporting both labels and annotations for backward compatibility.
+func ResolveTargetsFromIngressCallback(ingress *networkingv1.Ingress) ([]string, error) {
+	if ingress == nil {
+		return nil, errors.New("Ingress is nil")
+	}
+
+	if resourceLabelName, exists := ingress.Labels[common.NuclioResourceLabelKeyFunctionName]; exists {
+		if canaryLabelName, exists := ingress.Labels[common.NuclioResourceLabelKeyCanaryFunctionName]; exists {
+			return []string{resourceLabelName, canaryLabelName}, nil
+		}
+		return []string{resourceLabelName}, nil
+	}
+
+	// try to get the targets from the ingress annotations for backward compatibility
+	if annotation, exists := ingress.Annotations[common.NginxConfigurationSnippetAnnotationKey]; exists {
+		targets, err := extractTargetsFromAnnotation(annotation)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to extract resource names from the configuration snippet annotation")
+		}
+		if len(targets) == 0 {
+			return nil, errors.New("No resource names found in the nginx configuration snippet annotation")
+		}
+
+		return targets, nil
+	}
+
+	return nil, errors.New("Failed to resolve ingress resource names, no function name label or annotation found")
```

Comment:
it is already done here in https://github.com/v3io/scaler/  , why do we need to copy this logic?